federation/ldap/src/main/java/org/keycloak/storage/ldap/mappers/membership/group/GroupLDAPStorageMapper.java

```diff
@@ -74,6 +77,7 @@ public GroupLDAPStorageMapper(ComponentModel mapperModel, LDAPStorageProvider ld
         super(mapperModel, ldapProvider);
         this.config = new GroupMapperConfig(mapperModel);
         this.factory = factory;
+        session.getKeycloakSessionFactory().register(this);
```

Comment:
Ideally, we should register singletons as a listener. In this case, we can register the listener at the `postInit` method of the corresponding provider factory of this mapper.